tests/components/zeroconf/test_init.py

```diff
@@ -764,9 +790,28 @@ async def test_async_detect_interfaces_setting_empty_route(hass, mock_async_zero
     ), patch(
         "homeassistant.components.zeroconf.AsyncServiceInfo",
         side_effect=get_service_info_mock,
+    ), patch(
+        "socket.if_nametoindex",
+        side_effect=lambda iface: {"eth0": 1, "eth1": 2, "eth2": 3, "vtun0": 4}.get(
+            iface, 0
+        ),
     ):
         assert await async_setup_component(hass, zeroconf.DOMAIN, {zeroconf.DOMAIN: {}})
         hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
         await hass.async_block_till_done()
+    assert mock_zc.mock_calls[0] == call(
+        interfaces=[1, "192.168.1.5", "172.16.1.5", 3], ip_version=IPVersion.All
```

Comment:
We should always announce the ipv4 address first since some mdns implementations (mainly homekit accessories) will take the first address record they see.  These ones are less likely to support ipv6 so putting the ipv4 first should preserve compatibility.

Reply:
This highlighted line is the list of IP addresses or interface IDs for the constructor of `class Zeroconf` to configure multicast membership, not the address list for announcement. Testing the address list for announcement is done by a separate test case `test_get_announced_addresses`.

I have moved the logic to always put the existing ipv4 address as the first address into `_get_announced_addresses`.

Reply:
BTW I did some investigation and realized if we want to announce only the IP addresses of the interface where the mDNS queries come from in a multi-homed host, as https://datatracker.ietf.org/doc/html/rfc6762#section-14 written,  we need to change the implementation of `python-zeroconf`. I would like to investigate that direction in the next step.

Reply:
Some of what needs to happen to make that possible is outlined here:

https://github.com/jstasiak/python-zeroconf/issues/806